src/ape/managers/query.py

```diff
@@ -38,10 +38,61 @@ def perform_query(self, query: QueryType) -> pd.DataFrame:
 
     @perform_query.register
     def perform_block_query(self, query: BlockQuery) -> pd.DataFrame:
-        blocks_iter = self.chain_manager.blocks.range(query.start_block, query.stop_block)
+        blocks_iter = self.range(query.start_block, query.stop_block)
         block_dicts_iter = map(partial(get_columns_from_item, query), blocks_iter)
         return pd.DataFrame(columns=query.columns, data=block_dicts_iter)
 
+    def range(
```

Comment:
I think the name of this method has to change if we do this...

`DefaultQueryProvider.range()` does not indicate it returns blocks necessarily.